### DISCUSSION TITLE
Update 10.09.24

### DISCUSSION THREAD

**Discussion by s0cr4tes** [wasiejen/Free-Snap-Tap#44] [General]
Hey can you check after update today, 09.09.24, if the Code in still valid?

https://t.co/bNjCQXJRY3

Says theyre been adding some kind of detection.

**Comment by wasiejen** [comment]
Hiho,

I already played after the update today and had no problems so far. (I always play on valve servers)
And if I understand the Release Notes correctly they are just rolling out the counter-strafing detection from last month to non-valve servers. 
No mention of further or changed detection.

my current version of counter-strafing is this:

```bash
+w|(tr("+w")>50), !s, !ctrl, !space  :  +w|15|5, -s|(cs("+w")*1.25), +s|0|0
+s|(tr("+s")>50), !w, !ctrl, !space  :  +s|15|5, -w|(cs("+s")*1.25), +w|0|0
+a|(tr("+a")>50), !d, !ctrl, !space  :  +a|15|5, -d|(cs("+a")*1.25), +d|0|0
+d|(tr("+d")>50), !a, !ctrl, !space  :  +d|15|5, -a|(cs("+d")*1.25), +a|0|0
```

With this spamming counter-strafing is really hard because it only executes if a movement key was pressed for at least 50 ms. And it will not execute if the opposite key is pressed, already crouching or jumping. :-) Snap Tap works independent of that.

**Reply by stvnkng** [reply]
where should i paste this? in fst config?

**Reply by wasiejen** [reply]
yes in the fst config file :-D
but since version 1.0.0 the notation for macros has changed from using `:` to `::`

```bash
+w|(tr("+w")>50), !s, !ctrl, !space  ::  +w|15|5, -s|(cs("+w")*1.25), +s|0|0
+s|(tr("+s")>50), !w, !ctrl, !space  ::  +s|15|5, -w|(cs("+s")*1.25), +w|0|0
+a|(tr("+a")>50), !d, !ctrl, !space  ::  +a|15|5, -d|(cs("+a")*1.25), +d|0|0
+d|(tr("+d")>50), !a, !ctrl, !space  ::  +d|15|5, -a|(cs("+d")*1.25), +a|0|0
```

Should do the trick. or see the big example at the start of the ReadMe - there is very similar, if not the same version to be found.

right now i use:

```bash
# Macros
+w|(tr("+w")>100)|(last("-space")>500), !s, !ctrl ::  +w|15|5, -s|(cs("+w")*1.1), +s|0|0
+s|(tr("+s")>100)|(last("-space")>500), !w, !ctrl ::  +s|15|5, -w|(cs("+s")*1.1), +w|0|0
+a|(tr("+a")>100)|(last("-space")>500), !d, !ctrl ::  +a|15|5, -d|(cs("+a")*1.1), +d|0|0
+d|(tr("+d")>100)|(last("-space")>500), !a, !ctrl ::  +d|15|5, -a|(cs("+d")*1.1), +a|0|0
```
